pkg/controller/labelidentity/label_group_index.go

```diff
@@ -299,58 +297,42 @@ func (i *LabelIdentityIndex) deleteSelector(selectorKey string, policyKey string
 	}
 }
 
-// SetPolicySelectors registers ClusterSet-scope policy selectors with the labelIdentityIndex,
-// and then retrieves all the LabelIdentity IDs that currently match these selectors.
-func (i *LabelIdentityIndex) SetPolicySelectors(selectors []*types.GroupSelector, policyKey string) []uint32 {
-	var labelIdentityIDs []uint32
-	newSelectors := map[string]*types.GroupSelector{}
-	for _, s := range selectors {
-		klog.V(4).InfoS("Getting matched LabelIdentity for policy selector", "selector", s.NormalizedName, "policy", policyKey)
-		newSelectors[s.NormalizedName] = s
-	}
+// RemoveStalePolicySelectors cleans up any outdated selector <-> policy mapping based on the policy's latest selectors.
+func (i *LabelIdentityIndex) RemoveStalePolicySelectors(selectorKeys sets.String, policyKey string) {
 	originalSelectors := i.getPolicySelectors(policyKey)
-	for selKey, sel := range newSelectors {
-		if _, exists := originalSelectors[selKey]; exists {
-			// These clusterset-scoped selectors are already bound to the policy in labelIdentityIndex.
-			// We can simply read matched label identity IDs from the index.
-			selectedLabelIDs := i.GetLabelIdentityIDs(selKey)
-			labelIdentityIDs = append(labelIdentityIDs, selectedLabelIDs...)
-			// Remove matched clusterset-scoped selectors of the policy before and after the update.
-			// The selectors remaining in originalSelectors will need to be removed from the policy.
-			delete(originalSelectors, selKey)
-		} else {
-			selectedLabelIDs := i.AddSelector(sel, policyKey)
-			labelIdentityIDs = append(labelIdentityIDs, selectedLabelIDs...)
+	if originalSelectors == nil {
+		return
+	}
+	if selectorKeys.Len() > 0 {
+		for _, selKey := range selectorKeys.UnsortedList() {
```

Comment:
```suggestion
		for selKey := range selectorKeys {
```
no need to allocate a new slice

Reply:
Done